src/Definition/Common.php

```diff
@@ -15,20 +15,29 @@ class Common
     /**
      * @var bool
      */
-    protected $nullable = false;
+    protected $nullable;
+    /**
+     * @var string
+     */
+    protected $default;
 
     /**
      * Common constructor.
      *
      * @param $type
-     * @param string $length
-     * @param bool $nullable
+     * @param array $options -- length, nullable, default
      */
-    public function __construct($type, $length = null, $nullable = true)
+    public function __construct($type, $options = [])
     {
         $this->type = $type;
-        $this->length = $length;
-        $this->nullable = (bool) $nullable;
+        $this->length = (isset($options['length'])) ? $options['length'] : "";
+        $this->nullable = (isset($options['nullable'])) ? (bool) $options['nullable'] : true;
+        $this->default = "";
+        if (isset($options['default'])) {
+            $this->default = $options['default'];
+        } else if ($this->isNullable()) {
+            $this->default = "NULL";
```

Comment:
what distinguishes this from `"NULL"`?

eg

VARCHAR(50) DEFAULT NULL
vs
VARCHAR(50) DEFAULT "NULL"